demos/demo_create_solo12_robot.cpp

```diff
@@ -122,7 +122,12 @@ int main(int argc, char **argv)
             // Checks if the robot is in error state (that is, if any component
             // returns an error). If there is an error, the commands to send
             // are changed to send the safety control.
-            robot->SendCommand();
+            if (!robot->SendCommand())
+            {
+                throw std::runtime_error(
+                    "Error when sending the command. Please check error "
+                    "messages.");
+            }
```

Comment:
I am not sure throwing here is the right thing to do. When you throw an exception, the program gets terminated, the connection to the master-board gets closed and the robot goes into zero-torque mode. This means the robot just colaps in place (when using this for Solo12 for instance).

However, the `robot->SendCommand()` applies the safety controller, which does a high D gain joint damping, to slow down the robot and bring it slowly to stand. This only happens when the control loop is still running and `robot->SendCommand()` is called.

Do you see my point?

Reply:
Yes I see your point. You are right, I forgot to take into account how the robot will react if we directly throw an exception that terminates the program. I guess I can just let it as it was before without the exception throwing and let the user manually end the program when they see that the robot is in a safe position after the security damping.